### DISCUSSION TITLE
essay/2024/01/05/coffee-with-coffee-developer

### DISCUSSION THREAD

**Discussion by giscus** [BenjaminKim/benjaminkim.github.io#156] [Comments]
# essay/2024/01/05/coffee-with-coffee-developer

진로/취업 상담 등의 요청이 간혹 왔습니다만 대부분 거절했습니다. 내가 뭐라고 진로를 상담해 주나. 그런 건 알아서 하는 거지. 조언을 해준다는 것이 부담스러웠습니다.

https://jeho.page/essay/2024/01/05/coffee-with-coffee-developer.html

**Comment by spearkkk** [comment]
개발이라는 것을 어떻게 하면 지속할 수 있을까요?
사실 제가 개발을 좋아했던 이유는 새로운 것을 습득하고 어떤 것을 만들거나 해결하고 이 끝에 오는 성취감때문인데,
5년차 이후부터 성취감만 생각하는 것이 아니라 제 업무의 결과가 성과에 반영되는지에 대한 고민도 같이 하게 되더라구요.
저 스스로도 참 간사?한게 제가 한 업무가 성과에 반영이 안되거나 안될 것 같은 상황이라면 업무를 진짜 단순하게 처리하는 듯이 하게 되더라구요. 더욱이 현재 상황에서 제가 성과를 위해서 열심히 한다고 해서 이를 알아주지 않을 것 같다는 제 철없는 생각에 더 업무를 주도적으로 안하고 되구요.
이 상황에 계속되어서 업무(개발)에 대한 성취감이 결여가 되었고 더 이상 흥미를 못느끼고 있습니다.
어떻게 하면 업무를 통해 성취감을 느낄 수 있을까요?

**Reply by BenjaminKim** [reply]
열심히 했는데 성과나 연봉에 반영이 안되면 굉장히 속상하죠.
저도 그런 경험이 있습니다.
(언젠가 블로그에 써봐야겠네요)

알아주지 않는다고 해서 대충 일해버리면 결과는 정해져있습니다.
스스로를 망치는 것입니다.
남들이 알아주는데는 자기 생각보다 두배는 더 오래 걸린다고 생각해도 좋습니다.
어차피 본인을 위해서 일하는 것이니 남들을 의식하지 말고 최선을 다하셨으면 좋겠고..
그렇게 하다보면 지금 회사에서 분명 인정을 받게 되거나..
그걸 알아봐주는 훨씬 좋은 회사에서 일하게 될거라 생각합니다.

**Reply by spearkkk** [reply]
답변 감사합니다.

**Comment by yoonjong-park** [comment]
@BenjaminKim 
수익 1차 목표를 얼마 정도로 잡아두어야 할까요?
월급 정도로 잡으면 될지..
어느 정도 수익이 나면, 회사를 관두고 자기 일에 몰두하는 게 낫겠다고 생각하시는지 궁금합니다.
지금 80전후로 수익이 나고 있는 상태입니다.

**Reply by BenjaminKim** [reply]
목표로 얼마를 잡을지는 사람의 기질마다 다른 것 같습니다.
월급 정도 벌기 시작하면 마음이 편해지는 것은 분명합니다.
(월 80만 원 이익을 낸다는 것은 대단한 일입니다)

월수익으로만 판단하는 것보다는..
회사에서 배울 것이 아직 많이 남아있는지..
회사 생활이 얼마나 행복한지.. 혼자 일하는 것은 그에 비해 얼마나 행복한지..
사업은 얼마나 지속될 것 같은지 등..
여러 변수를 종합적으로 고려해서 결정하시면 좋을 것 같습니다.

**Reply by yoonjong-park** [reply]
@benjaminKim 
답글 감사드립니다.
참고해서 고민해보겠습니다.
간간히 소식 남기겠습니다.

**Reply by BenjaminKim** [reply]
네, 소식 전해주세요!

**Reply by yoonjong-park** [reply]
@BenjaminKim 수익을 내는 방법을 좀 바꾸었더니, 매출이 갑자기 많이 늘고 있습니다.
재호님도 수익이 없다가 갑자기 느는 지점이 있으셨는지요?
이럴 때 제가 주의해야 할 것은 무엇인지 조언 좀 구하고 싶습니다.

**Reply by BenjaminKim** [reply]
와.. 축하드립니다.
저도 물론 그럴 때가 있었습니다.
주의해야 할 점은 지금과는 완전히 반대로... 잘 나가다가 갑자기 수익이 줄 때도 있다는 것.
언제든지 떨어질 수 있다는 생각을 하고 일을 하면 마음도 편하고 일도 잘 되는 것 같습니다.

**Reply by yoonjong-park** [reply]
네. 기대가 욕심이 되지 않도록 주의하면서 정진해보겠습니다.
조언 감사드립니다.😆

**Comment by hyunsangwon** [comment]
안녕하세요, 
블로그 소개 중 '전업 투자자이기도 합니다' 글 보고 질문 남겨봅니다😊

이제 6년 차구요~! 올해 32이네요
한참 코인, 부동산 열풍일 때 저는 개발에만 집중했습니다..
투자 공부할 시간에 개발 공부에 투자하자는 마인드였습니다만.. 이제 결혼과 미래를 생각하며
투자는 선택이 아닌 필수라는 걸 좀 늦게 알았습니다.
이제야 투자 공부를 최근 들어 하고 있습니다... (그렇다고 개발공부를 멈춘 건 아닙니다)

조금 더 일찍했을 걸 이라는 후회감이 요새 들더라구요..
심지어 첫 회사는 코인회사였습니다.. 퇴직금을 이더리움으로 준다고 했는데 거절했죠..(당시 19년도)
인생에서 가장 후회스러운 일입니다 ㅎ...
참, 투자도 공부할께 끝이 없더라구요 ㅎㅎ... 모든게 산 넘어 산인 것 같네요!

재호 님도 상당한 개발덕후?!로 보이는데, 예전부터 투자를 천천히 공부하신 건지 아니면 
은퇴 후 투자 공부에 집중하신 건지 궁금합니다. 그리고 추천할 책 있으면 감사하겠습니다 ㅎㅎ

**Reply by BenjaminKim** [reply]
개발에만 집중하는 것은 좋은 일이라 생각합니다.
본업을 잘하는 것이 무엇보다 먼저입니다.

저는 33살 때까지는 투자에는 관심이 없었습니다.
은퇴 후에 투자 관련한 공부들을 했는데... 이게 저에게 그렇게 즐겁지는 않습니다.
주식창에서 멀어져서 살려고 애쓰고 있습니다.

26살의 사회초년생 저에게 투자에 대해서 말할 수 있는 기회가 생긴다면...
월급 남는 것은 매달 s&p를 사고 투자 같은 것에는 신경쓰지 말고 코딩이나 열심히 하고 살라고 말해주고 싶습니다만...
26살의 저는 아마 그 말을 듣지 않았을 것 같네요.(웃음)

책은... 월가의 영웅하고 워런 버핏 바이블 추천합니다.

**Reply by hyunsangwon** [reply]
빠른 답변 감사합니다 ㅎㅎ 
영화 인터스텔라처럼 과거 저에게 말할 기회가 있다면
매달 남는 돈으로 술먹지말고 S&P나 금 적금하라고 할 것 같네요 🤣

다시 한번 답변 감사합니다 ^^
24년 새해 복 많이 받으시구 앞으로 블로그 글 기대하겠습니다 :)

**Comment by kyugorithm** [comment]
안녕하세요, 
올려주신 1인 개발자에 대한 글을 읽고 관심이 많이 갔던 사람입니다. 재밌는 글 기고해 주셔서 감사해요 ^^

AI 분야에 있긴 했지만 분야를 뒤늦게 바꿔서 생성 AI 연구를 시작하게 됐고 쌓아 둔 것이 많지 않아 마음이 조급해지는것을 피할 수 없는것 같습니다. (그래서 더 많은 시간을 사용하다보니 가족과의 시간도 자연스레 줄어들게 된것 같습니다. ㅠㅠ)
그리고 기술 발전 속도가 너무 빠르다 보니 50대 넘어서까지 어떤 회사에 소속되어서 이 일을 할 수 있을까 하는 걱정도 많이 듭니다. 

나중엔 재호님처럼 1인 개발자로 가족과 더 가까이 지내면서 자유롭게 생활하고싶은 마음이 크네요..
그러기 위해 엔지니어링과 프론트 개발 등등을 공부하면서 사이드 프로젝트도 진행해 보고 이것저것 열심히 뭔가 하고 있긴 합니다..

처음에 어떤 계기로 1인 개발자가 되기로 마음먹게 되셨는지 궁금하고 빠르게 변화하는 기술이나 여러가지의 속도에 부담은 없
으신지 궁금합니다.
또 하고계신 아이템에 대한 아이디어는 어떻게 생각하게 되셨는지도 궁금합니다.

감사해요!

**Reply by BenjaminKim** [reply]
궁금해하시는 점들을 적은 글들입니다.
저도 빨리 변하는 기술들이 물론 부담스럽지만.. 조급해하지 않으려 마음을 다잡고 있습니다.

* [딥러닝과 블록체인](https://jeho.page/essay/2022/09/05/deep-learning-and-blockchain.html)
* [회사를 그만두고 1인 개발을 해야 할까요?](https://jeho.page/essay/2023/07/14/should-i-quit-company.html)
* [왜 하필 소개팅 어플을 만들기로 했어요?](https://jeho.page/essay/2021/09/07/%EC%99%9C-%ED%95%98%ED%95%84-%EC%86%8C%EA%B0%9C%ED%8C%85-%EC%96%B4%ED%94%8C%EC%9D%84-%EB%A7%8C%EB%93%A4%EA%B8%B0%EB%A1%9C-%ED%96%88%EC%96%B4%EC%9A%94.html)

**Reply by kyugorithm** [reply]
답변 주셔서 감사합니다. ^^ 
언젠가 좋은기회로 교류할 기회가 있으면 좋겠습니다.
새해 복많이 받으셔요 ~!

**Comment by kimyow** [comment]
안녕하세요. 커피한 잔 앱, 웹, github, blog 까지 관련 환경을 잘 구축해 놓으셨군요.
유튜브 영상으로 보고, 막연히 대단하다고 생각했었는데, 여러 곳에서 작성하신 아티클들을 보니 새삼 대단하다고 느껴집니다.

저는 대기업에 재직중이고, 별도 서비스 출시해서 나름 수익을 내고 있는데요.
한 가지 문의 드릴 것이 있습니다.
1인 개발자이시니, 앱과 백엔드 모두 개발하실 수 있는 역량은 물론, 운영, 영업, 마케팅까지 모두 경험해 보셨을텐데요.
"커피한잔 외에 다른 앱을 추가 개발하면 좀 더 수익이 나지 않을까?" 라는 욕심이 생기지 않으셨을까요?
혼자서도 충분히 개발하여 런칭까지 하실 수 있을텐데요.
서비스 다작을 해서 좀 더 수익을 만들어볼까 라는 욕심은 안 드셨는지 궁금합니다.
아니면, 이 정도면 되겠다고 생각하시는 어떤 계기가 있었을까요?
현재의 삶을 만족하고 계시다면, 개인적으로 어떤 목표를 가지고 생활하시는지도 궁금하네요.

**Reply by BenjaminKim** [reply]
조금이라도 젊을 때 더 만들어 보고 싶은 욕심이 있습니다.
다만 서비스 하나 제대로 만드는 게 워낙 힘들고 피곤한 일이기에 섣불리 손대지 못하고 있을 뿐...
그래도 계속 열심히 해봐야죠.
목표는 [해외에서 작은 돈이라도](https://jeho.page/essay/2023/09/29/give-back-more-than-take.html) 벌어서 세금으로 내는 것입니다.

**Reply by kimyow** [reply]
오~ 목표(사회에 기여하는 방법)가 공감이 가네요^^
답변 감사합니다.

**Comment by rosy83** [comment]
안녕하세요.

Youtube 통해서 동영상(조코딩) 보고, 여기까지 와서 고민 상담을 합니다.
저는 바이오 제약 분야에서 국내 10년 이상, 국외 3년 가량 경험이 있습니다.
전문 분야는 신약개발 관리/ 및 바이오 공정 인데요.

제가 작년 겨울부터 1인 창업을 하고, 웹 포탈 소프트웨어 (대상은 바이오쪽 분석/비임상 시험  기관 고객사이며, 고객수가 국내 100개 이하입니다. 매우 적은 고객을 타겟)을 제작하고 있습니다.
주변에 IT쪽에 아는 사람이 없어서 인터넷으로 찾아서 제품 개발 문의했습니다.

첫번째 제작 결과가 시제품으로 내놓을 수가 없어, 최근에 2번째 의뢰를 하였는데요.
제시한 금액이 첫번째 제작 금액의 거의 2배였습니다.
제가 작년 겨울에 퇴사를 해서 빨리 제품을 출시해서 영업해야 하는 마음으로 우선 제작을 하기로 했는데요.

지금 너무 불안합니다.
또 제대로 안 만들어 주면 어쩌지 하는 마음과 시장 가격 대비 너무 많은 금액을 지불한 것이 아닌가 하는 마음입니다. (B2B 계약이라 볼 수 있으나, 실상 저는 개인사업자로 현재 저의 모아둔 개인 자금으로 지출)

담당자의 이메일 답신은 빠른데, 소규모 업체 그 쪽 개발하시는 분은 다른 분인 건지 오탈자 등의 수정 피드백을 줘도 수 주일이 걸리는 실정입니다.

저의 제품은 소프트웨어를 이용하고자 하는 고객사와 계약을 하게 되면, 시제품을 일부 커스터마이징 해서 (고객을 확보할 때마다 개발사에게 견적 의뢰를 해야함.) 판매 할 예정이고 중간에 유지 보수도 개발사에 의뢰해 해야 하는 실정이라, 개발사에게 지속적으로 의뢰를 해야 하는데, 현재 상황으로 비지니스 파트너의 관계를 유지하는 것이 어렵겠다는 판단이 섭니다.

우선 1차 제작의 제작 수준이 제가 지금 모든 것을 걸고 하는 일을 하찮게 여기는 것이 아닌가 하는 수준이고요.
앞으로 커스터마이즈 할 때마다 가격을 또 시장가 보다 상당히 높게 되면, 실제 제가 밖에서 발로 뛰면서 마음 졸이며 영업하는데, 개발비 주고 나면, 결국에 남는 이익을 없지 않을까 우려스럽습니다.

이에 대한 상담을 하고 싶습니다. 

감사합니다.

**Reply by rosy83** [reply]
2번째 제작을 위해 지급한 금액은 천만원대 입니다. 소프트웨어가 결과관리 프로그램으로, 단순 결과 입력과 대시보드와 챗팅 기능은 있으나, 다른 프로그램과 연동하거나 하지 않습니다.

**Reply by rosy83** [reply]
2번째 제작은 1차 제작품의 수정 보완 버전입니다.

**Reply by BenjaminKim** [reply]
안녕하세요.
저도 개발자를 고용해서 일을 해본 적이 없어 상담을 해드리거나 도움을 드릴 수가 없습니다.
오픈된 시장에서 좋은 개발자를 구한다는 것은... 좋은 변호사나 의사를 구하는 것보다 훨씬 어려운 일일 것 같습니다.
그러니 차라리 내가 배워서 하는 게 빠르겠다라는 말이 나오는 거겠죠.
아무쪼록 잘 해결되길 바라겠습니다.

**Reply by pabami** [reply]
아... 저와 비슷한 부분이 있어서 답글 남겨요.
전 교육분야가 본업이고 경력 13년차입니다. 83년 생이고...
작은 시장을 타겟으로 해서(최대 200개) 전 제가 독학으로 만들었어요... 시간이 오래 걸렸긴 하지만요.
직접 만든 이유는... 의뢰해서 만들면 정말 비쌉니다. 제가 만든 프로그램이 1-3억 한다고 하더라구요.(아직 그만큼 수익은 못 냈지만...) 작은 시장이라 투자도 받기 어렵고요.

1천만원 상당으로는 양산형 쉬운 것 만드는 것 아니면 퀄리티가 정말 낮을 거에요. 신뢰할만한 찾기도 어렵고요.

본업 놔두고 사비로 개발 의뢰해서 하시느라 마음이 바짝바짝 마르실 것 같아요. ㅠㅠ

손익을 잘 계산하셔서 사업 방향을 정해지시고 잘 해결되길 바랍니다

**Comment by voidFunctionReturn0** [comment]
안녕하세요. 2부 리그 서비스기획자입니다.
취미로 개발 프로젝트를 하고 있습니다.
블로그 글을 통해 영감을 많이 받고 있습니다.
재호 님을 롤모델로 삼을 수 있고 정진하고 있습니다.
이렇게 꾸준히 블로그 글을 써주시는 것만으로도 영광입니다.
감사합니다.

**Reply by voidFunctionReturn0** [reply]
삼을 수 있고 -> 삼아

입니다 ㅎㅎ 수정 기능이 없네용

**Comment by voidFunctionReturn0** [comment]
공개 가능하시다면, 커피한잔 서비스 운영 비용이 어느 정도인지 궁금합니다!

제 이메일은 voidfunctionretirn0@gmail.com
입니다.

**Reply by BenjaminKim** [reply]
커피한잔 서버 운영에 대해서는 언젠가 블로그 포스트로 써보려합니다. 😁

**Reply by BenjaminKim** [reply]
서버 운영 관련 경험을 아래 글들로 정리했습니다.
* [홈 서버 운영의 장점](https://jeho.page/essay/2024/04/29/home-server.html)
* [홈 서버 운영의 단점](https://jeho.page/essay/2024/05/03/coffee-server.html)

유튜브에서도 장애났던 이야기를 자세하게 했네요. 현재의 비용 이야기도 다 말해버렸습니다. ㅎ
https://www.youtube.com/watch?v=Z2VXtzFYd1w

**Comment by pabami** [comment]
감사합니다. 이메일로 커피챗 신청했습니다. ^^

**Reply by pabami** [reply]
소중한 시간 내어 주셔서 정말 감사합니다. 하는 일에 더 자신감을 갖게 되었습니다. 꼭 다시 연락드릴께요. ^^

**Reply by BenjaminKim** [reply]
저도 즐거운 시간이었습니다.
열심히 코딩하시고.. 연락 기다리겠습니다! 💪

**Comment by SungchangChoi** [comment]
안녕하세요. 1인 개발자를 목표로하고 있습니다. 
만들려고 하는 앱이 사진을 업로드 하는 기능이 있습니다. 
그래서 조코딩에서 재호선배님께서 말씀하신대로 집에다 우선 서버를 두고 운영하다가, 운이 좋아 사용자가 늘어나면 AWS 든 다른 서버리스를 이용할까합니다. 제가 지금까지 플런트 엔드만 해봐서 서버를 만들어 본적이 없는데, 2~3년 하면 서버를 뚝딱 만들 수 있다고 하셔서 도전해보고자합니다. 
어떤 공부를 하면 좋을지라던가 재호선배님께서 서버를 어떻게 만드셨는지 알려주실 수 있으신가요?

제 이메일은 nangchoi@gmail.com 입니다. 
미리 답변 감사합니다.

**Reply by BenjaminKim** [reply]
백엔드 코스 과정이 이미 잘 정리된 곳이 많으니 한 번 찾아보시면 될 것 같습니다.
저 같은 경우는 Ruby on Rails 로 백엔드 거의 모든 과정을 처리하고 있어요.
